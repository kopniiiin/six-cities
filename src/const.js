export const OfferType = {
  APARTMENT: `apartment`,
  HOTEL: `hotel`,
  HOUSE: `house`,
  ROOM: `room`
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const DEFAULT_CITY = City.PARIS;

export const SortType = {
  POPULAR: `popular`,
  TO_HIGH_PRICE: `price: low to high`,
  TO_LOW_PRICE: `price: high to low`,
  TOP_RATED: `top rated first`
};

export const DEFAULT_SORT_TYPE = SortType.POPULAR;

export const ServerURL = {
  LOGIN: `/login`,
  OFFERS: `/hotels`
};

export const ServerResponseStatus = {
  OK: 200,
  UNAUTHORIZED: 401
};

export const AuthorizationStatus = {
  AUTHORIZED: `AUTHORIZED`,
  UNAUTHORIZED: `UNAUTHORIZED`
};
