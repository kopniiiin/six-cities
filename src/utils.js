const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`
];

export const extend = (target, update) => Object.assign({}, target, update);

export const pushElement = (array, element) => [...array, element];

export const replaceElement = (array, element, index) => [...array.slice(0, index), element, ...array.slice(index + 1)];

export const removeElement = (array, index) => [...array.slice(0, index), ...array.slice(index + 1)];

export const upperCaseFirstLetter = (string) => string[0].toUpperCase() + string.slice(1);

export const formatDate = (dateString) => {
  const date = new Date(dateString);
  return `${MONTHS[date.getMonth()]} ${date.getDate()}, ${date.getFullYear()}`;
};
