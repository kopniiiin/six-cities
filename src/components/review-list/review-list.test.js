import React from "react";
import {shallow} from "enzyme";

import {AuthorizationStatus} from "../../const.js";

import {extend} from "../../utils.js";

import ReviewList from "./review-list.jsx";

import testMocks from "../../test-mocks/review-list.js";

describe(`snapshot test: ReviewList component`, () => {
  it(`should render correctly`, () => expect(
      shallow(<ReviewList {...testMocks}/>)
  ).toMatchSnapshot());

  it(`should render ReviewForm`, () => expect(
      shallow(<ReviewList {...extend(testMocks, {authorizationStatus: AuthorizationStatus.AUTHORIZED})}/>)
  ).toMatchSnapshot());
});
