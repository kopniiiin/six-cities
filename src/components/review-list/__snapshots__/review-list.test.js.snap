// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: ReviewList component should render ReviewForm 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <Review
      date="2020-04-04"
      id="4"
      key="4"
      rating={4}
      text="Good"
      user={
        Object {
          "name": "User",
          "photo": "photo",
        }
      }
    />
  </ul>
  <WithReviewData
    onSubmit={[Function]}
  />
</section>
`;

exports[`snapshot test: ReviewList component should render correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews · 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <Review
      date="2020-04-04"
      id="4"
      key="4"
      rating={4}
      text="Good"
      user={
        Object {
          "name": "User",
          "photo": "photo",
        }
      }
    />
  </ul>
</section>
`;
